app/src/main/java/com/plenart/organizeme/fragments/MembersFragment.kt

```diff
@@ -0,0 +1,158 @@
+package com.plenart.organizeme.fragments
+
+import android.app.Dialog
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.Observer
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.plenart.organizeme.adapters.MemberListItemAdapter
+import com.plenart.organizeme.databinding.DialogAddSearchMemberBinding
+import com.plenart.organizeme.databinding.FragmentMembersBinding
+import com.plenart.organizeme.viewModels.MembersViewModel
+import java.lang.reflect.InvocationTargetException
+
+
+class MembersFragment : Fragment() {
+    private lateinit var binding: FragmentMembersBinding
+    private val viewModel: MembersViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        binding = FragmentMembersBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initObservers()
+        initListeners()
+        try {
+            getArgs()
+        } catch (e: InvocationTargetException) {
+            Log.e("memfr", e.cause.toString())
+        }
+
+        viewModel.getAssignedMembersListDetails()
+
+    }
+
+    private fun getArgs() {
+        val args: MembersFragmentArgs by navArgs()
+        viewModel.setBoardDetails(args.boardDetails)
+
+    }
+
+    private fun initListeners() {
+        binding.fabMember.setOnClickListener {
+            dialogAddSearchMember();
+        }
+    }
+
+
+    private fun initObservers() {
+        initAssignedMembers()
+        initMemberAssigned()
+        initMember()
+    }
+
+    private fun initMember() {
+        viewModel.member?.observe(viewLifecycleOwner, Observer {
+            memberDetails()
```

Comment:
Why you don't pass user through memberDetails() method instead getting value of user inside memberDetails. This is not good aproach.

-> `viewModel.member.observe(viewLifecycleOwner){ user ->
            memberDetails(user)
       }`